include/TPP/Dialect/Xsmm/XsmmOps.td

```diff
@@ -140,6 +141,34 @@ def Xsmm_UnaryOp : Xsmm_Op<"unary", []> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// MatmulOp
+//===----------------------------------------------------------------------===//
+
+def Xsmm_MatmulOp : Xsmm_Op<"matmul"> {
+  let summary = "matmul call operation.";
+  let arguments = (ins Xsmm_DataType:$dataType, Variadic<XsmmMemRef>:$inputs);
```

Comment:
I don't think we should be using `Variadic` here, since the number of arguments doesn't change.

This may have less human-written boilerplate, but it passes the wrong idea to the compiler and to whoemever reads this later. It wouldn't be a lot more code to have a clear number of arguments for each op type.

Reply:
Yes, but this would require custom builders as the assembly format dictates the builder signature. I can do it but this will PR will likely grow even more.